### DISCUSSION TITLE
session.get("https://auth.tesla.com/oauth2/v3/authorize", headers=headers, params=params)  never returns in AWS EC2

### DISCUSSION THREAD

**Discussion by labdel** [timdorr/tesla-api#329] [Q&A]
Hi,
I am running the pyhton version of the code in https://github.com/enode-engineering/tesla-oauth2/blob/main/tesla.py.
It runs perfect in my local PC pycharm, I am able to get the token and refresh token. Problem I have is when I run same .py file in my AWS EC2 ubuntu machine it never returns 

session.get("https://auth.tesla.com/oauth2/v3/authorize", headers=headers, params=params) 

I appeciate your feedback.

Thanks in advance - Leo.

**Comment by briangru** [comment]
When running on my local dev machine, adding User-agent headers seemed to cause the Tesla API calls to return successfully.  
I will not guess about the running in AWS vs. locally differences - there may be something else going on.

**Reply by labdel** [reply]
Hi briangru,
guess what..... they are banning traffic from AWS for the authentication only. So it seems it is a matter of just redirecting traffic throug some other server/machine. Need to try that.

**Reply by mehdirahbe** [reply]
Did you manage to make it work? I am stuck too with heroku app (Aws hosted). Proxy did not work anymore...

**Reply by Noy** [reply]
I am also facing similar issues, having the login page load normally on my local but as soon as I'm on any linux based system on AWS, Vultr or DigitalOcean, I'm getting a forbidden error accessing auth.tesla.com - is there a way around this? Unfortunately user agent headers aren't working
